### DISCUSSION TITLE
lazyvim doesn't work in nixos?

### DISCUSSION THREAD

**Discussion by cStralpt** [LazyVim/LazyVim#4925] [Q&A]
i found that lazyvim has problems with nixos, like the LSP won't run. the only possible solutions is just to wrap the executable with steam-run which is not possible with the hundreads of executable that needs to wrapped manually.
if you guys know any alternative that really close with lazyvim please let me know or if you guys know how to solve this problems, it would be very help full thanks.

**Comment by ferid333** [comment]
Hello @cStralpt 

LazyVim on NixOS can face issues because some dependencies aren’t available in Nix's sandboxed environment. Wrapping each executable with `steam-run` isn't practical.

**Solutions:**
1. **Alternative Configs:** Check out [[AstroNvim](https://astronvim.com/)](https://astronvim.com) or [[NvChad](https://nvchad.com/)](https://nvchad.com), which offer similar preconfigured setups and may work better on NixOS.
2. **Fix LazyVim:** Use `nixpkgs` to install required LSP servers and dependencies globally or per-project via `flake.nix`. This avoids wrapping executables manually.

**Reply by cStralpt** [reply]
i think nvchad is great too, but i'll figure out how to setuo lazyvim using nixvim first, if no chance i'll rethink to switch to nvchad. thanks for the suggestions

**Comment by Jacobus-afk** [comment]
Hi @cStralpt, have you read through this [discussion](https://github.com/LazyVim/LazyVim/discussions/1972)? I've found that by following the way @azuwis set up his config, got me a working lazyvim

It does use home manager, don't know if that might be a no-no for you?

**Reply by cStralpt** [reply]
thanks for the swift reply, i will try it out fast!

**Reply by cStralpt** [reply]
@folke is there any plan to officially support nixos in the future update of lazyvim?

**Reply by folke** [reply]
I don't use Nixos, so not by me
